### DISCUSSION TITLE
Should `{% fragment ... %}` be in wagtailcore tags?

### DISCUSSION THREAD

**Discussion by SebCorbin** [wagtail/wagtail#12135] [Development]
We have a project structured so that `wagtail.admin` is not installed in the frontoffice, and we were wondering why the `fragment` template tag is not in wagtail core, instead of wagtail admin? It's pretty useful, kinda like the one from `django-slippers`

**Comment by thibaudcolas** [comment]
Hi @SebCorbin, it’s because at the point of adding it, it was only intended for usage within the admin of the CMS. If people find it useful outside the CMS, I think that’d make for a solid enhancement. Open a feature request in the issue tracker?

Generally speaking, I don’t think we’d want to introduce too many tags or filters that aren’t directly related to Wagtail’s features, but if we have such template customizations that are already in use in the admin, might as well make them available more widely if there are clear use cases.

**Comment by gasman** [comment]
This sort of thing creates a bit of a maintenance headache for us - it means we're on the hook for supporting this tag indefinitely in user code, even if we move to a different way of achieving the same thing in the Wagtail admin itself.

According to the docstring, Wagtail's `fragment` tag _is_ a copy-paste from django-slippers, so why not just use django-slippers? I suspect the only reason we copy-pasted it into wagtail.admin rather than adding django-slippers as a dependency was to avoid users having to update INSTALLED_APPS in their projects - although that's avoidable by importing the tag into wagtailadmin_tags, as we did for the `component` tag in `laces`.

**Reply by thibaudcolas** [reply]
Seems to me like "use Slippers" or "here is the code to copy-paste into your project" or "update to our new way of doing things" would be fine approaches during deprecation?

I copy-pasted it to avoid the dependency altogether, seeing this is currently the only part of the library we’ve considered using.

**Comment by SebCorbin** [comment]
I hear you two and I'm wondering, why not propose this enhancement in Django at all?

**Reply by gasman** [reply]
Because that's a whole load of coding and advocacy work that I don't feel particularly invested in :-) I wouldn't object to someone else giving it a go, though.

**Reply by laymonage** [reply]
FYI it's been discussed here, though it didn't seem to go very far: https://forum.djangoproject.com/t/adding-template-fragments-or-partials-for-the-dtl/21500

Carlton has made django-template-partials since then https://github.com/carltongibson/django-template-partials, though it works differently.

**Reply by thibaudcolas** [reply]
Yes I was part of that discussion, I think i’s a fine enhancement proposal for Django, don’t have the time / energy to propose it myself at this point.

**Reply by SebCorbin** [reply]
Wish me luck https://code.djangoproject.com/ticket/6378#comment:31

**Reply by SebCorbin** [reply]
FYI `{% simple_block_tag %}` landed [in core](https://docs.djangoproject.com/en/5.2/releases/5.2/#templates)

**Reply by thibaudcolas** [reply]
Excellent!

**Reply by SebCorbin** [reply]
Furthermore as [wagtailadmin is required](https://github.com/wagtail/wagtail/issues/12271), I think we can close this.
